packages/beacon-node/src/chain/chain.ts

```diff
@@ -476,7 +484,7 @@ export class BeaconChain implements IBeaconChain {
 
   async produceCommonBlockBody(blockAttributes: BlockAttributes): Promise<CommonBlockBody> {
     const {slot} = blockAttributes;
-    const head = this.forkChoice.getHead();
+    const head = blockAttributes.proposerHead ?? this.forkChoice.getHead();
```

Comment:
I don't see the change in `chain.produceBlockWrapper()` where you use `proposerHead` inside `BlockAttributes` to build block